src/main/java/axoloti/IPlatform.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (C) 2013 - 2016 Johannes Taelman
+ *
+ * This file is part of Axoloti.
+ *
+ * Axoloti is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Axoloti is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * Axoloti. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Mark Harris
+ */
+package axoloti;
+
+import axoloti.object.AxoObject;
+
+/**
+ *
+ * @author Mark Harris
+ */
+public interface IPlatform {
```

Comment:
the point of interfaces is they are that, a declaration of contract.
PlatformBase is used as an implementation vehicle for methods which will be common to many platforms. Ive a few other methods that will go on here.
the difference is Platforms will be able to just implement the interface, there is no requirement for the to derive from PlatformBase. 
(Its why I didn't call it Platform, as that would seem like it has to be in the hierarchy)